src/main/java/org/gridsuite/study/server/networkmodificationtree/dto/BuildStatus.java

```diff
@@ -7,11 +7,31 @@
 
 package org.gridsuite.study.server.networkmodificationtree.dto;
 
+import org.gridsuite.study.server.dto.modification.NetworkModificationResult;
+
 /**
+ * The order of the enum is important and is used to know if we need to update a status.
+ * Values priority is based on their ordinal (higher ordinal => higher priority).
  * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
  */
 public enum BuildStatus {
     NOT_BUILT,
+    BUILDING,
     BUILT,
-    BUILDING;
+    BUILT_WITH_WARNING,
+    BUILT_WITH_ERROR;
+
+    public static BuildStatus fromApplicationStatus(NetworkModificationResult.ApplicationStatus status) {
+        if (status.equals(NetworkModificationResult.ApplicationStatus.WITH_ERRORS)) {
+            return BuildStatus.BUILT_WITH_ERROR;
+        } else if (status.equals(NetworkModificationResult.ApplicationStatus.WITH_WARNINGS)) {
+            return BuildStatus.BUILT_WITH_WARNING;
+        } else {
+            return BuildStatus.BUILT;
+        }
+    }
+
+    public boolean isBuilt() {
+        return ordinal() >= 2;
```

Comment:
Did not verify, but could we use BUILT.ordinal() in stead of plain 2 ?

Reply:
Good idea, I change it